src/main/java/xyz/aadev/generitech/common/tileentities/power/TileEntityPower.java

```diff
@@ -0,0 +1,196 @@
+package xyz.aadev.generitech.common.tileentities.power;
+
+
+import mcp.mobius.waila.api.IWailaConfigHandler;
+import mcp.mobius.waila.api.IWailaDataAccessor;
+import net.darkhax.tesla.api.ITeslaConsumer;
+import net.darkhax.tesla.api.ITeslaHolder;
+import net.darkhax.tesla.api.ITeslaProducer;
+import net.darkhax.tesla.api.implementation.BaseTeslaContainer;
+import net.darkhax.tesla.capability.TeslaCapabilities;
+import net.darkhax.tesla.lib.TeslaUtils;
+import net.minecraft.block.Block;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import xyz.aadev.aalib.api.common.integrations.waila.IWailaBodyMessage;
+import xyz.aadev.aalib.client.util.LanguageHelper;
+import xyz.aadev.aalib.common.inventory.InternalInventory;
+import xyz.aadev.aalib.common.inventory.InventoryOperation;
+import xyz.aadev.aalib.common.tileentities.TileEntityInventoryBase;
+import xyz.aadev.generitech.Reference;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class TileEntityPower extends TileEntityInventoryBase implements ITeslaProducer, net.minecraft.util.ITickable, IWailaBodyMessage {
+    private BaseTeslaContainer container = new BaseTeslaContainer(0, 50000, 1000, 1000);
+    private InternalInventory inventory = new InternalInventory(this, 1);
+    private int T0transfer = 100;
+    private int fuelRemaining = 0;
+    private Item lastFuelType;
+    private int lastFuelValue;
+
+    @Override
+    public long takePower(long power, boolean simulated) {
+        return power;
+    }
+
+
+
+    @Override
+    public void update() {
+        BlockPos pos = getPos();
+        World worldIn = getWorld();
+
+        if (fuelRemaining!=0)fuelRemaining--;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)